userspace/libscap/scap_bpf.c

```diff
@@ -140,6 +141,22 @@ static int bpf_map_create(enum bpf_map_type map_type,
 	return sys_bpf(BPF_MAP_CREATE, &attr, sizeof(attr));
 }
 
+static int bpf_map_freeze(int fd)
+{
+	/* BPF_MAP_FREEZE was introduced in kernel 5.2 */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,2,0)
+	union bpf_attr attr;
+
+	bzero(&attr, sizeof(attr));
+
+	attr.map_fd = fd;
+
+	return sys_bpf(BPF_MAP_FREEZE, &attr, sizeof(attr));
+#else
+	return SCAP_SUCCESS;
+#endif
+}
+
```

Comment:
I'm not sure we should do a kernel version check here. This chooses the code path based on the kernel on the machine used to build this code. As it's userspace, it can be easily shared between hosts running different kernels.

IMO we should just go ahead with the syscall and either ignore all errors, or a specific subset based on the actual syscall returns; we probably want to ignore ENOSYS, maybe also EINVAL and/or EPERM (if we run into a runc with a broken seccomp filter)

Reply:
Thanks @gnosek  for your input.
As there is no great benefit apart from "being on the safe side", i guess we can just ignore all errors.

Reply:
I removed latest commit and force-pushed, to revert to previous implementation.

Reply:
(I suggested doing it this way, sorry for the conflicting feedback, you can leave it this way, I guess).

Reply:
no problem sir :)